src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -874,53 +868,43 @@ public without sharing class BDI_MigrationMappingUtility {
-                        this.targetFieldAPIName =
-                            UTIL_Namespace.alignSchemaNSWithEnvironment(this.targetFieldAPIName);
-                    }
+                List<String> mappingValuesFromHelpText = retrieveObjectAndFieldApiNames(this.helpText);
+                if (mappingValuesFromHelpText != null) {
+                    this.dataImportObjectName = mappingValuesFromHelpText[0];
+                    this.targetFieldAPIName =
+                        UTIL_Namespace.shouldAlignNamespace ?
+                            UTIL_Namespace.alignSchemaNSWithEnvironment(mappingValuesFromHelpText[1]) :
+                            mappingValuesFromHelpText[1];
 
                     checkRequiredFields();
                 }
             }
         }
 
-        private List<String> splitHelpText(String helpText) {
-            String symbol;
-            Boolean hasSpecialChar = helpText.containsAny('$&+,:;=?@#|\'<>.-^*()%!_');
-            Boolean hasDataImportNames =
-                helpText.containsAny('Account1Account2AddressContact1Contact2HouseholdOpportunityDonationPaymentGAURole');
-
-            if (hasSpecialChar == true) {
-                String[] specialCharacters =
-                    new List<String>{'$', '&', '\\+', ',', ':', ';', '=', '\\?', '@',
-                                     '#', '\\|', '\'', '<', '>', '\\.', '-', '^', '\\*',
-                                     '\\(', '\\)', '%', '!', '_'};
-
-                for (String sc : specialCharacters) {
-                    List<String> parts = helpText.split(sc, 2);
-                    Integer countWhitespaces = helpText.countMatches(' ');
-
-                    if (parts.size() >= 2 && countWhitespaces <= 4) {
-                        if (String.isNotBlank(parts[0]) && String.isNotBlank(parts[1])) {
-                            return parts;
-                        }
+        /*******************************************************************************************************
+        * @description Retrieves an object and field api name from a DataImport__c field help text mapping.
+        *
+        * @param helpText Help text value of a DataImport__c field
+        * @return A list of strings containing an object api name and a field api name OR null if help text
+        * value doesn't contain a valid help text mapping.
+        */
+        private List<String> retrieveObjectAndFieldApiNames(String helpText) {
+            if (helpText.contains('.')) {
+                List<String> parts = helpText.split('\\.', 2);
+
+                if (parts.size() == 2) {
+                    String possibleObjectApiName = parts[0];
+                    String possibleFieldApiName = parts[1];
+                    Boolean isLegacyObjectApiName =
+                        LEGACY_DATA_IMPORT_OBJECT_NAMES.contains(possibleObjectApiName.toLowerCase());
+                    Boolean isPossibleFieldApiName = !possibleFieldApiName.contains(' ');
```

Comment:
Can help text have leading or trailing whitespace? I wonder if we should toss a ".trim()" in here just to avoid nixing a possible mapping because it has a space at the end of it